### PR TITLE
5.next: add plugin versions to `plugin list` command

### PR DESCRIPTION
**Pull request by LordSimal**
Resolves https://github.com/cakephp/cakephp/issues/17541

I have adjusted the PluginConfig class to also contain the logic to parse the `composer.json` for the package name as well as the `composer.lock` to get the exact version which is being used for a specific package.

So this can easily be reused to create a `bin/cake plugin show DebugKit` command if so desired (or adjust the already existing packages panel of debug_kit)

Using this branch on my current app this currently looks like this:
```
+-------------------------+-----------+------------+----------+----------+---------+
| Plugin                  | Is Loaded | Only Debug | Only CLI | Optional | Version |
+-------------------------+-----------+------------+----------+----------+---------+
| ADmad/Sequence          | X         |            |          |          | 4.0.0   |
| AlfredCovid             | X         |            |          |          |         |
| AlfredDashboard         | X         |            |          |          |         |
| AlfredDbIntegrity       | X         |            |          |          |         |
| AlfredDeploy            | X         |            |          |          |         |
| AlfredDns               | X         |            |          |          |         |
| AlfredDocumentations    | X         |            |          |          |         |
| AlfredDriversLogbook    | X         |            |          |          |         |
| AlfredEasybill          | X         |            |          |          |         |
| AlfredGoogle            | X         |            |          |          |         |
| AlfredMaintenances      | X         |            |          |          |         |
| AlfredMealOrders        | X         |            |          |          |         |
| AlfredProjects          | X         |            |          |          |         |
| AlfredStaffMembers      | X         |            |          |          |         |
| AlfredStandBy           | X         |            |          |          |         |
| AlfredTemplates         | X         | X          |          |          |         |
| AlfredTools             | X         |            |          |          |         |
| Authentication          |           |            |          |          | 3.0.3   |
| Authorization           |           |            |          |          | 3.1.1   |
| Bake                    | X         |            | X        | X        | 3.0.5   |
| CakeDC/Auth             |           |            |          |          | 8.0.3   |
| CakeDC/Users            | X         |            |          |          | 12.0.0  |
| CakeDumpSql             | X         |            |          |          | 2.0.0   |
| CakeSentry              | X         |            |          |          | 3.x-dev |
| Cake/TwigView           | X         |            |          |          | 2.0.1   |
| CakephpFixtureFactories | X         |            | X        | X        | v3.0.1  |
| CakephpTestSuiteLight   |           |            |          |          | v3.0    |
| DebugKit                | X         | X          |          |          | 5.0.5   |
| Duplicatable            | X         |            |          |          | 5.0.0   |
| IdeHelper               | X         |            | X        | X        | 2.2.0   |
| IdeHelperExtra          | X         |            | X        | X        | 0.2.0   |
| Migrations              | X         |            | X        | X        | 4.1.1   |
| Muffin/Footprint        | X         |            |          |          | 4.0.1   |
| Queue                   | X         |            |          |          | 7.1.2   |
| Search                  | X         |            |          |          | 7.0.0   |
| Setup                   | X         | X          |          |          | 3.2.1   |
| Shim                    |           |            |          |          | 3.2.1   |
| Tools                   | X         |            |          |          | 3.3.0   |
| LordSimal/CakephpDbml   |           |            |          |          |         |
+-------------------------+-----------+------------+----------+----------+---------+
```

we could of course add another column at the end to see if that specific plugin/package is a base requirement or a dev requirement in the `composer.json` but I guess this isn't that much of a usefull information so I left it out.

Same goes for the composer package name

### CODE REVIEW

**Comment by ADmad** [comment]
I hope none of this additional composer config files parsing is run during normal plugin loading.

**Comment by LordSimal** [comment]
> I hope none of this additional composer config files parsing is run during normal plugin loading.

Currently only the PluginListCommand uses the `PluginConfig::getAppConfig` method and therefore the file parsing.

In a future DebugKit Version it will also use this for the Packages Tab but thats also expected I would say.

But no, this is not being used in the normal Plugin bootstrapping workflow.

File: src/Command/PluginListCommand.php [hunk 1]
```diff
@@ -85,6 +88,9 @@ public function execute(Arguments $args, ConsoleIo $io): ?int
     public function buildOptionParser(ConsoleOptionParser $parser): ConsoleOptionParser
     {
         $parser->setDescription('Displays all currently available plugins.');
+        $parser->addOption('composer-path', [
```
Comment (ADmad) [inline src/Command/PluginListCommand.php:91]: Is this option necessary? Isn't the `composer.lock` always going to be under the `ROOT` path?

Comment (LordSimal) [reply src/Command/PluginListCommand.php:91]: I primarily added this option to make testing easier.

**Pull request merged**